frontstage/controllers/collection_exercise_controller.py

```diff
@@ -88,8 +89,83 @@ def convert_events_to_new_format(events):
             "month": date_time.strftime('%m'),
-            "is_in_future": date_time > parse_date(datetime.now().isoformat())
+            "is_in_future": date_time > parse_date(datetime.now().isoformat()),
+            "formatted_date": ordinal_date_formatter('{S} %B %Y, %H:%M', date_time),
+            "due_time": due_date_convertor(date_time)
         }
     return formatted_events
+
+
+def suffix(day: int):
+    """
+    This function creates the ordinal suffix
+
+    :param: day of the date time object
+    :return: ordinal suffix
+    """
+    return 'th' if 11 <= day <= 13 else {1: 'st', 2: 'nd', 3: 'rd'}.get(day % 10, 'th')
+
+
+def ordinal_date_formatter(date_format_required: str, date_to_be_formatted: datetime):
+    """
+    This function takes the required output format and date to be formatted and returns the ordinal date in required
+    format.
+
+    :param: date_format_required: output format in which date should be returned
+    :param: date_to_be_formatted: the datetime object which needs ordinal date
+    :return: formatted date
+    """
+    return date_to_be_formatted. \
+        strftime(date_format_required).replace('{S}', str(date_to_be_formatted.day) + suffix(date_to_be_formatted.day))
+
+
+def due_date_convertor(date: datetime):
+    """
+    This function provides the custom due date based on the difference between now and the date passed.
+    The logic for the due date is based on the following.
+    Due date is today - Due today
+    Due date is tomorrow - Due tomorrow
+    Due date is 2-28 days - Due in X days
+    Due date is 29-59 days - Due in a month
+    Due date is 60-89 days - Due in 2 months
+    Due date is 90-119 days - Due in 3 months
+    Due date is 120+ days - Due in over 3 months
+
+    :param: date: the datetime date for which due date is to be evaluated.
+    :return: due date
+    """
+    now = datetime.now()
+    event_day = datetime(date.year, date.month, date.day)
+    today = return_date_time(now)
+    tomorrow = return_date_time(now + timedelta(days=1))
+    day_after = return_date_time(now + timedelta(days=2))
+    a_month = return_date_time(now + timedelta(days=29))
+    two_months = return_date_time(now + timedelta(days=60))
+    three_months = return_date_time(now + timedelta(days=90))
+    four_months = return_date_time(now + timedelta(days=120))
+    if event_day == today:
+        return 'Due today'
+    if event_day == tomorrow:
+        return 'Due tomorrow'
+    if day_after <= event_day < a_month:
+        delta = date.replace(tzinfo=None) - now.replace(tzinfo=None)
+        return f'Due in {delta.days} days'
+    if a_month <= event_day < two_months:
+        return 'Due in a month'
+    if two_months <= event_day < three_months:
+        return 'Due in 2 months'
+    if three_months <= event_day < four_months:
+        return 'Due in 3 months'
+    if four_months <= event_day:
+        return 'Due in over 3 months'
+
+
+def return_date_time(timedelta_now: datetime):
```

Comment:
Pretty minor, but the signature can also include the return value.
```python
def return_date_time(timedelta_now: datetime) -> datetime:
```

Reply:
👍